### PR TITLE
[#2151] Update CSS-related Major Dependencies

### PR DESCRIPTION
**Pull request by sopa301**
Part of #2151.

### Proposed commit message

```
Update CSS-related Major Dependencies

A few CSS-related dependencies have major updates available.

Let's update as many as possible.
```

### Other information

`highlight.js` has version `11` available, but I've avoided updating it because of [breaking changes](https://github.com/highlightjs/highlight.js/blob/main/VERSION_11_UPGRADE.md#feature-removal) concerning unescaped HTML that the current (outdated) Pug loader cannot fix easily. Fortunately, there is a replacement [`@webdiscus/pug-loader`](https://github.com/webdiscus/pug-loader) that can both replace the deprecated `pug-plain-loader` and ([in the future](https://webdiscus.github.io/pug-loader/pug-filters/highlight.html)) implement Pug-friendly `highlight.js` without the need for an external library.

### CODE REVIEW

**Review by jonasongg — APPROVED** [review]
LGTM!

File: frontend/package.json [hunk 1]
```diff
@@ -42,8 +42,8 @@
   },
   "devDependencies": {
     "@babel/eslint-parser": "^7.24.1",
-    "@typescript-eslint/eslint-plugin": "^6.0.0",
-    "@typescript-eslint/parser": "^6.0.0",
+    "@typescript-eslint/eslint-plugin": "^5.5.0",
+    "@typescript-eslint/parser": "^5.5.0",
```
Comment (ckcherry23) [inline frontend/package.json:46]: Is there any reason these were downgraded?

Comment (sopa301) [reply frontend/package.json]: Whoops! That was unintended, I have re-added those.

**Review by ckcherry23 — APPROVED** [review]
LGTM!

**Review by MarcusTXK — APPROVED** [review]
LGTM

**Comment by ckcherry23** [comment]
Hi @sopa301, can you please help resolve the merge conflicts so that we can merge this?

**Comment by sopa301** [comment]
Done!

**Pull request merged**